### PR TITLE
integration-tests: Changes to make it work with live networks again

### PR DESCRIPTION
**Pull request by mslipper**

### CODE REVIEW

File: integration-tests/test/setup-docker-compose-network.ts [hunk 1]
```diff
@@ -1,7 +0,0 @@
-import { DockerComposeNetwork } from './shared/docker-compose'
```
Comment (tynes) [inline integration-tests/test/setup-docker-compose-network.ts:1]: Why did you delete this files? To be honest, I never used them but I'm not sure if other people did. I always run `docker compose` manually

File: integration-tests/test/shared/env.ts [hunk 2]
```diff
@@ -83,8 +85,10 @@ export class OptimismEnv {
 
     // fund the user if needed
     const balance = await l2Wallet.getBalance()
-    if (balance.lt(utils.parseEther('1'))) {
-      await fundUser(watcher, l1Bridge, utils.parseEther('1').sub(balance))
+    const min = envConfig.L2_WALLET_MIN_BALANCE_ETH.toString()
```
Comment (tynes) [inline integration-tests/test/shared/env.ts:88]: Confirmed there is a default value

File: integration-tests/test/shared/stress-test-helpers.ts [hunk 3]
```diff
@@ -23,7 +23,7 @@ export const fundRandomWallet = async (
   const fundTx = await env.l1Wallet.sendTransaction({
     gasLimit: 25_000,
     to: wallet.address,
-    gasPrice: await gasPriceForL1(env),
+    gasPrice: await gasPriceForL1(),
```
Comment (tynes) [inline integration-tests/test/shared/stress-test-helpers.ts:26]: Gas estimation on L1 is failing? That is pretty strange to be honest

File: integration-tests/test/shared/utils.ts [hunk 4]
```diff
@@ -175,56 +241,23 @@ export const defaultTransactionFactory = () => {
...
+export const gasPriceForL2 = async () => {
+  if (procEnv.L2_GAS_PRICE === ON_CHAIN_GAS_PRICE) {
+    return l2Wallet.getGasPrice()
   }
 
-  if (isLiveNetwork()) {
-    return Promise.resolve(BigNumber.from(10000))
-  }
-  return Promise.resolve(BigNumber.from(0))
+  return utils.parseUnits(procEnv.L2_GAS_PRICE, 'gwei')
```
Comment (tynes) [inline integration-tests/test/shared/utils.ts:249]: Shouldn't this be `wei`?

File: integration-tests/test/shared/utils.ts [hunk 5]
```diff
@@ -175,56 +241,23 @@ export const defaultTransactionFactory = () => {
...
-    case 5:
-      return utils.parseUnits('2', 'gwei')
-    default:
-      return BigNumber.from(0)
+export const gasPriceForL1 = async () => {
+  if (procEnv.L1_GAS_PRICE === ON_CHAIN_GAS_PRICE) {
+    return l1Wallet.getGasPrice()
   }
+
+  return utils.parseUnits(procEnv.L1_GAS_PRICE, 'gwei')
```
Comment (tynes) [inline integration-tests/test/shared/utils.ts:257]: Same as above, this should be `wei`

File: integration-tests/test/replica.spec.ts [hunk 6]
```diff
@@ -4,26 +4,26 @@ import {
...
+  envConfig,
 } from './shared/utils'
 import { TransactionReceipt } from '@ethersproject/abstract-provider'
 
 describe('Replica Tests', () => {
   let env: OptimismEnv
 
-  before(async () => {
+  before(async function () {
+    if (!envConfig.RUN_DEBUG_TRACE_TESTS) {
```
Comment (tynes) [inline integration-tests/test/replica.spec.ts:15]: Should this be `RUN_REPLICA_TESTS`?

**Comment by tynes** [comment]
It looks like the L2 gasprice is being set to 0 in some places now based on the itest logs

```
Error: processing response error (body="{\"jsonrpc\":\"2.0\",\"id\":1114,\"error\":{\"code\":-32000,\"message\":\"cannot accept 0 gas price transaction\"}}\n", error={"code":-32000}, requestBody="{\"method\":\"eth_sendRawTransaction\",\"params\":[\"0xf868268082520894a8bd13ea2daadc1f0f1db036c67ac4d0fbc99cff872386f26fc100008082036ba081234402acea5122d30d7e955e3ee91466fbad724094de51f66b6e189b504b4aa02669e4b028fb671cd68fc0eec26909d28f511ee52f2acadc7aa4d8a395730657\"],\"id\":1114,\"jsonrpc\":\"2.0\"}", requestMethod="POST", url="http://l2geth:8545", code=SERVER_ERROR, version=web/5.4.0)
```

**Comment by tynes** [comment]
Could you add a changeset as well?

**Comment by tynes** [comment]
Also looks like some unused deps

```
Unused devDependencies
* @types/shelljs
* docker-compose
* shelljs
```

**Comment by mslipper** [comment]
This is still a bit of a WIP - working through some additional issues and needed it to run on CI.

File: integration-tests/test/shared/stress-test-helpers.ts (hunk 3 above)
Comment (mslipper) [reply integration-tests/test/shared/stress-test-helpers.ts:26]: This just lets us specify a gas price manually if we want.

File: integration-tests/test/setup-docker-compose-network.ts (hunk 1 above)
Comment (mslipper) [reply integration-tests/test/setup-docker-compose-network.ts:1]: I don't think that these should be part of the repo. It introduces a hard coupling between the tests we define here, and the environment that they run on. It also causes the CLI to do a bunch of unexpected stuff unless an env var is set.

File: integration-tests/test/fee-payment.spec.ts [hunk 7]
```diff
@@ -28,24 +25,25 @@ describe('Fee Payment Integration Tests', async () => {
...
+    `should return eth_gasPrice equal to OVM_GasPriceOracle.gasPrice`,
+    async () => {
       const assertGasPrice = async () => {
         const gasPrice = await env.l2Wallet.getGasPrice()
         const oracleGasPrice = await env.gasPriceOracle.gasPrice()
         expect(gasPrice).to.deep.equal(oracleGasPrice)
       }
 
-      assertGasPrice()
+      await assertGasPrice()
```
Comment (tynes) [inline integration-tests/test/fee-payment.spec.ts:37]: Good catch here

**Review by tynes — APPROVED** [review]
